gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/model/Operation.java

```diff
@@ -155,16 +155,28 @@ public void apply(ObjectNode node) {
             JsonNode parentNode = node.get(getPath().getAttributePath());
             if (getPath().getSubAttribute() != null) {
                 if (parentNode != null) {
-                    ((ObjectNode) parentNode).set(getPath().getSubAttribute(), value);
+                    if (parentNode.isArray() && MultiValuedAttributes.isListOfObjects(getPath().getAttributePath())) {
+                        fillObjectAttributes(((ArrayNode) parentNode).addObject());
+                    } else {
+                        ((ObjectNode) parentNode).set(getPath().getSubAttribute(), value);
+                    }
                 } else {
-                    node.putObject(getPath().getAttributePath()).set(getPath().getSubAttribute(), value);
+                    if (MultiValuedAttributes.isListOfObjects(getPath().getAttributePath())) {
+                        fillObjectAttributes(node.putArray(getPath().getAttributePath()).addObject());
+                    } else {
+                        node.putObject(getPath().getAttributePath()).set(getPath().getSubAttribute(), value);
+                    }
                 }
             } else if (parentNode == null) {
-                node.set(getPath().getAttributePath(), value);
+                if (MultiValuedAttributes.isListOfSingular(getPath().getAttributePath())) {
```

Comment:
why isn't the function fillObjectAttributes used here?

Reply:
This is because Singular value are not complex object, this is either a simple string or numberic.